src/UI/Badge.elm

```diff
@@ -195,88 +249,156 @@ successDark content =
 
 -}
 withTone : (String -> Badge) -> Badge -> Badge
-withTone builder (Badge { content }) =
-    builder content
-
+withTone builder (Badge { content } originalOptions) =
+    let
+        (Badge newProps _) =
+            builder content
+    in
+    Badge newProps originalOptions
 
 
--- Render
+companionIcon : RenderConfig -> Icon -> Color -> Element msg
+companionIcon cfg icon iconColor =
+    icon
+        |> Icon.withCustomSize 16
+        |> Icon.withColor iconColor
+        |> Icon.renderElement cfg
 
 
 {-| End of the builder's life.
 The result of this function is a ready-to-insert Elm UI's Element.
 -}
 renderElement : RenderConfig -> Badge -> Element msg
-renderElement cfg (Badge { content, tone, brightness }) =
-    let
-        ( background, textColor ) =
-            colors tone brightness
-    in
-    Text.subtitle2 content
-        |> Text.withColor textColor
-        |> Text.withOverflow Text.ellipsize
-        |> Text.renderElement cfg
-        |> Element.el
-            [ Element.width shrink
-            , Font.center
-            , background
-                |> Palette.toElementColor
-                |> Background.color
-            , Element.paddingEach { top = 6, bottom = 6, left = 8, right = 8 }
-            , Element.height (px 24)
-            , Border.rounded 12
-            ]
-
-
-colors : BadgeTone -> BadgeBrightness -> ( Palette.Color, Palette.Color )
+renderElement cfg (Badge { content, tone, brightness } { icon }) =
+    case icon of
+        Just i ->
+            let
+                ( background, textColor, outlineColor ) =
+                    colors tone brightness
+            in
+            Element.row
+                [ Element.width shrink
+                , Font.center
+                , background
+                    |> Palette.toElementColor
+                    |> Background.color
+                , outlineColor
+                    |> Palette.toElementColor
+                    |> Border.color
+                , Element.height (px 24)
+                , Border.rounded 12
+                , Border.width 2
+                ]
+                [ Element.column
+                    [ Element.paddingEach { top = 2, bottom = 2, left = 3, right = 2 }
+                    ]
+                    [ companionIcon cfg i textColor ]
```

Comment:
As it's a single element, use `Element.el`, and the same below...

Reply:
Fixed, thanks for pointing out :)